### PR TITLE
Create seeds

### PR DESCRIPTION
**Pull request by KoubaFromPucks**
Seeds are done. If you think it is not much data i can add some + tell me the case that is not implemented here (for example if there was no question of type multichoice).
Cases:
- 2 users (first has no picture, second user has pictureUrl)
- 2 Quizzes (first has 3 questions, second has 4 questions)
- There is 7 questions of all types. Some has just text, some has just picture and some has both.
- There are 3 completed quizzes. Somwhere is skipped question, some answers are ok, some answers are wrong.

### CODE REVIEW

File: src/Quizzer.API.DAL/Seeds/Storage.cs [hunk 1]
```diff
@@ -0,0 +1,559 @@
...
+                Id = QuestionGuids[1],
+                Type = Enums.AnswerFormat.SingleChoice,
+                Text = "Who is the king of great britain?",
+                QuizId = QuizGuids[0]
+            });
+
+            Questions.Add(new QuestionEntity {
+                Id = QuestionGuids[2],
+                Type = Enums.AnswerFormat.SingleChoice,
+                Text = "Is Austraia part of British commonwealth?",
```
Comment (brtkkndrl) [inline src/Quizzer.API.DAL/Seeds/Storage.cs]: typo

File: src/Quizzer.API.DAL/Seeds/Storage.cs [hunk 2]
```diff
@@ -0,0 +1,559 @@
...
+            });
+
+            UserAnswers.Add(new UserAnswerEntity {
+                Id = UserAnswerGuids[7],
+                AnswerTime = DateTime.Parse("18.01.2023 19:48", System.Globalization.CultureInfo.CreateSpecificCulture("fr-FR")),
+                UserId = UserGuids[1],
+                QuestionId = QuestionGuids[5]
+            });
+
+            UserAnswers.Add(new UserAnswerEntity {
```
Comment (brtkkndrl) [inline src/Quizzer.API.DAL/Seeds/Storage.cs:512]: Selected Answers don't need to be seeded here?

**Review by brtkkndrl — APPROVED** [review]

File: src/Quizzer.API.DAL/Seeds/Storage.cs [hunk 3]
```diff
@@ -0,0 +1,559 @@
...
+                SeedUserAnswers();
+                SeedSelectedAnswers();
+            }
+        }
+
+        private void SeedQuizzes() {
+            Quizzes.Add(new QuizEntity {
+                Id = QuizGuids[0],
+                StartTime = DateTime.Parse("15.01.2023 00:00", System.Globalization.CultureInfo.CreateSpecificCulture("fr-FR")),
+                EndTime = DateTime.Parse("15.01.2024 19:34", System.Globalization.CultureInfo.CreateSpecificCulture("fr-FR")),
```
Comment (vecerap) [inline src/Quizzer.API.DAL/Seeds/Storage.cs:158]: Why there is french culture?

**Review by vecerap — COMMENTED** [review]
The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?

**Comment by brtkkndrl** [comment]
> The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?

I agree.

**Comment by KoubaFromPucks** [comment]
> The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?

I am not sure if i understand you well. But you are saying that i should name the ids? So there would not be a list of ids but all ids would have a name like QuestionA, QuestionB... (in code, there will be `QuestionA = new Guid(...);` ) Is that the point?

File: src/Quizzer.API.DAL/Seeds/Storage.cs (hunk 2 above)
Comment (KoubaFromPucks) [reply src/Quizzer.API.DAL/Seeds/Storage.cs:512]: No, it does not.

File: src/Quizzer.API.DAL/Seeds/Storage.cs (hunk 3 above)
Comment (KoubaFromPucks) [reply src/Quizzer.API.DAL/Seeds/Storage.cs:158]: I like this format and i wanted to be sure, that those datetimes will be parsed correctly.

File: src/Quizzer.API.DAL/Seeds/Storage.cs (hunk 1 above)
Comment (KoubaFromPucks) [reply src/Quizzer.API.DAL/Seeds/Storage.cs]: fixed

**Comment by vecerap** [comment]
> > The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?
> 
> I am not sure if i understand you well. But you are saying that i should name the ids? So there would not be a list of ids but all ids would have a name like QuestionA, QuestionB... (in code, there will be `QuestionA = new Guid(...);` ) Is that the point?

@KoubaFromPucks well yes, maybe make them like a property... it can be only getter, getting the specific ID at a given index in the id list

File: src/Quizzer.API.DAL/Seeds/Storage.cs (hunk 3 above)
Comment (vecerap) [reply src/Quizzer.API.DAL/Seeds/Storage.cs:158]: cs-cz does not have this format? I don't mind it that at all, but it surprised me choosing this culture.

**Comment by vecerap** [comment]
> > > The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?
> > 
> > 
> > I am not sure if i understand you well. But you are saying that i should name the ids? So there would not be a list of ids but all ids would have a name like QuestionA, QuestionB... (in code, there will be `QuestionA = new Guid(...);` ) Is that the point?
> 
> @KoubaFromPucks well yes, maybe make them like a property... it can be only getter, getting the specific ID at a given index in the id list

In code this could be `public Guid QuestionA => questionIds[2];`

**Comment by KoubaFromPucks** [comment]
> > > > The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?
> > > 
> > > 
> > > I am not sure if i understand you well. But you are saying that i should name the ids? So there would not be a list of ids but all ids would have a name like QuestionA, QuestionB... (in code, there will be `QuestionA = new Guid(...);` ) Is that the point?
> > 
> > @KoubaFromPucks well yes, maybe make them like a property... it can be only getter, getting the specific ID at a given index in the id list
> 
> In code this could be `public Guid QuestionA => questionIds[2];`

Should this be made for every id? Because then the id lists are not really needed. Right?

**Comment by KoubaFromPucks** [comment]
> > > > > The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?
> > > > 
> > > > 
> > > > I am not sure if i understand you well. But you are saying that i should name the ids? So there would not be a list of ids but all ids would have a name like QuestionA, QuestionB... (in code, there will be `QuestionA = new Guid(...);` ) Is that the point?
> > > 
> > > @KoubaFromPucks well yes, maybe make them like a property... it can be only getter, getting the specific ID at a given index in the id list
> > 
> > In code this could be `public Guid QuestionA => questionIds[2];`
> 
> Should this be made for every id? Because then the id lists are not really needed. Right?

And I guess that answer maybe really don't need this approach.

**Comment by vecerap** [comment]
> > > > > The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?
> > > > 
> > > > 
> > > > I am not sure if i understand you well. But you are saying that i should name the ids? So there would not be a list of ids but all ids would have a name like QuestionA, QuestionB... (in code, there will be `QuestionA = new Guid(...);` ) Is that the point?
> > > 
> > > 
> > > @KoubaFromPucks well yes, maybe make them like a property... it can be only getter, getting the specific ID at a given index in the id list
> > 
> > 
> > In code this could be `public Guid QuestionA => questionIds[2];`
> 
> Should this be made for every id? Because then the id lists are not really needed. Right?

@KoubaFromPucks Well, I will leave it up to you, what stays is with names, the tests verification will be easier.

**Comment by vecerap** [comment]
> > > > > > The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?
> > > > > 
> > > > > 
> > > > > I am not sure if i understand you well. But you are saying that i should name the ids? So there would not be a list of ids but all ids would have a name like QuestionA, QuestionB... (in code, there will be `QuestionA = new Guid(...);` ) Is that the point?
> > > > 
> > > > 
> > > > @KoubaFromPucks well yes, maybe make them like a property... it can be only getter, getting the specific ID at a given index in the id list
> > > 
> > > 
> > > In code this could be `public Guid QuestionA => questionIds[2];`
> > 
> > 
> > Should this be made for every id? Because then the id lists are not really needed. Right?
> 
> And I guess that answer maybe really don't need this approach.

I am not forcing you to do it on every id, make it where it makes sense later during the tests creation.

**Comment by KoubaFromPucks** [comment]
> > > > > > > The overall structure seems good, but what maybe is missing is some naming of the seed data. What I mean is when I compare it to the ICS project we had named seeds and during unit tests, we could easily refer to those specific seed data, here you have IDs where anyone could make a mistake, because it looks all similar. I don't encourage to make some names a totally valid could be letters (Question A, Question B, etc, or more specific names like skipped question A, B etc...) What do you think about this approach @BRTEK007 ?
> > > > > > 
> > > > > > 
> > > > > > I am not sure if i understand you well. But you are saying that i should name the ids? So there would not be a list of ids but all ids would have a name like QuestionA, QuestionB... (in code, there will be `QuestionA = new Guid(...);` ) Is that the point?
> > > > > 
> > > > > 
> > > > > @KoubaFromPucks well yes, maybe make them like a property... it can be only getter, getting the specific ID at a given index in the id list
> > > > 
> > > > 
> > > > In code this could be `public Guid QuestionA => questionIds[2];`
> > > 
> > > 
> > > Should this be made for every id? Because then the id lists are not really needed. Right?
> > 
> > 
> > And I guess that answer maybe really don't need this approach.
> 
> I am not forcing you to do it on every id, make it where it makes sense later during the tests creation.

Changes done

**Review by vecerap — APPROVED** [review]
Squash commit b97ae56359ae65cd5134942bdeab44262ab603f4 with 91d9834768a986a167acae8f6b9442bcc87c3e10. And keep message from 91d9834768a986a167acae8f6b9442bcc87c3e10

**Comment by KoubaFromPucks** [comment]
> Squash commit [b97ae56](https://github.com/vecerap/iw5-2023/commit/b97ae56359ae65cd5134942bdeab44262ab603f4) with [91d9834](https://github.com/vecerap/iw5-2023/commit/91d9834768a986a167acae8f6b9442bcc87c3e10). And keep message from [91d9834](https://github.com/vecerap/iw5-2023/commit/91d9834768a986a167acae8f6b9442bcc87c3e10)

done

**Pull request merged**